lib/jit/generation/gemm.cpp

```diff
@@ -2674,12 +2685,19 @@ std::string gemm::generate_impl(std::string const & suffix, expression_tree cons
   stream << "}" << std::endl;
 
   if(has_depth)
-  {
-    stream << "$KERNEL void reduce" << suffix << "($SIZE_T M, $SIZE_T N, $SIZE_T D, "
-           << "$GLOBAL " << sdtype << "* Z, $SIZE_T Zld,"
-           << "$GLOBAL " << sdtype << "* C, $SIZE_T ldc, $SIZE_T Cstart, $SIZE_T Cstride,"
-           << sdtype << " beta)"
-           << std::endl;
+  { if(sdtype !="half"){
+      stream << "$KERNEL void reduce" << suffix << "($SIZE_T M, $SIZE_T N, $SIZE_T D, "
+             << "$GLOBAL " << sdtype << "* Z, $SIZE_T Zld,"
+             << "$GLOBAL " << sdtype << "* C, $SIZE_T ldc, $SIZE_T Cstart, $SIZE_T Cstride,"
+             << sdtype << " beta)"
+             << std::endl;
+      }else{
+      stream << "$KERNEL void reduce" << suffix << "($SIZE_T M, $SIZE_T N, $SIZE_T D, "
+             << "$GLOBAL " << sdtype << "* Z, $SIZE_T Zld,"
+             << "$GLOBAL " << sdtype << "* C, $SIZE_T ldc, $SIZE_T Cstart, $SIZE_T Cstride,"
+             << "float" << " beta)"
+             << std::endl;
+      }
     stream << "{" << std::endl;
     stream.inc_tab();
 
```

Comment:
Probably these two branches for the kernel argument can be replaced by:

std::string abdtype = (sdtype=="half")?"float":sdtype